### PR TITLE
Add GUI is ready function and add it on visualization tests

### PR DESCRIPTION
**Pull request by carolhmj**
This should avoid the tests with GUI failing because it hasn't loaded yet.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by RaananW — APPROVED** [review]

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13475/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13475/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13475/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13475/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13475/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13475/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13475/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13475/merge#BCU1XR#0

**Review by sebavan — APPROVED** [review]

File: packages/dev/gui/src/2D/advancedDynamicTexture.ts [hunk 1]
```diff
@@ -47,6 +47,9 @@ export class AdvancedDynamicTexture extends DynamicTexture {
     /** Snippet ID if the content was created from the snippet server */
     public snippetId: string;
 
+    /** Observable that fires when the GUI is ready */
+    public onGuiReadyObservable = new Observable<AdvancedDynamicTexture>();
```
Comment (sebavan) [inline packages/dev/gui/src/2D/advancedDynamicTexture.ts:51]: Can we clear in dispose just to be safe if we never reach ready before dispose ???

File: packages/tools/testTools/src/visualizationUtils.ts [hunk 2]
```diff
@@ -232,9 +232,14 @@ export const evaluateRenderSceneForVisualization = async (renderCount: number) =
             if (window.scene.activeCamera && (window.scene.activeCamera as any).useAutoRotationBehavior) {
                 (window.scene.activeCamera as any).useAutoRotationBehavior = false;
             }
+            const sceneAdts: any[] = window.scene!.textures.filter((t: any) => t.getClassName() === "AdvancedDynamicTexture");
+            const adtsAreReady = () => {
```
Comment (sebavan) [inline packages/tools/testTools/src/visualizationUtils.ts:236]: should not gui is ready be part of scene ready ?

it is strange to check this here

**Pull request merged**